lib/datadog/lambda/utils/extension.rb

```diff
@@ -17,7 +17,17 @@ module Utils
     EXTENSION_CHECK_URI = URI(AGENT_URL + HELLO_PATH).freeze
     EXTENSION_PATH = '/opt/extensions/datadog-agent'
 
-    def self.extension_running
+    @is_extension_running = nil
+
+    def self.extension_running?
+      return @is_extension_running unless @is_extension_running.nil?
+
+      @is_extension_running = check_extension_running
+
+      @is_extension_running
```

Comment:
I'm guessing this means auto return the result of @is_extension_running, I wonder why we don't use return? is it just for brevity?

Reply:
It's the idiomatic way of doing it.
Referencing your previous comment: http://ruby-for-beginners.rubymonstas.org/writing_methods/return_values.html